src/material/table/testing/row-harness.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {RowHarnessFilters, CellHarnessFilters} from './table-harness-filters';
+import {MatCellHarness, MatHeaderCellHarness, MatFooterCellHarness} from './cell-harness';
+
+/** Data extracted from the cells in a table row. */
+export type MatRowHarnessData = {
+  columnName: string;
+  text: string;
+}[];
+
+/** Harness for interacting with a standard Angular Material table row. */
+export class MatRowHarness extends ComponentHarness {
+  static hostSelector = '.mat-row';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a row with specific attributes.
+   */
+  static with(options: RowHarnessFilters = {}): HarnessPredicate<MatRowHarness> {
+    return new HarnessPredicate(MatRowHarness, options);
+  }
+
+  /** Gets all cells of the table row. */
+  async getCells(filter: CellHarnessFilters = {}): Promise<MatCellHarness[]> {
+    return this.locatorForAll(MatCellHarness.with(filter))();
+  }
+
+  /** Gets the data of the cells in the footer row. */
+  async getData(filter: CellHarnessFilters = {}): Promise<MatRowHarnessData> {
```

Comment:
I don't like calling this `data` since people are going to mistakenly think it means that actual data object the row is bound against, which we don't have access to. Maybe for now it would be better to just let people do e.g.
```ts
(await rowHarness.getCells()).map(c => c.getColumnName());
```

Reply:
I feel like it is a useful method to have, what if we just call it something less ambiguous, e.g.: `getRowText`, `getCellsAsText`, `yourSuggestionHere`

Reply:
I also think that it's useful, especially the one that organizes it by columns since it requires a bit more logic to match the columns to the headers. I'll rename it to be less generic.